SysArchMeter.c

```diff
@@ -0,0 +1,74 @@
+/*
+htop - SysArchMeter.c
+(C) 2021 htop dev team
+Released under the GNU GPLv2, see the COPYING file
+in the source distribution for its full text.
+*/
+#include "config.h"  // IWYU pragma: keep
+
+#include "SysArchMeter.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <sys/utsname.h>
+
+#include "XUtils.h"
+
+
+static const int SysArchMeter_attributes[] = {HOSTNAME};
+
+static void SysArchMeter_updateValues(Meter* this, char* buffer, size_t size) {
+   static struct utsname uname_info;
+   static int uname_result;
+   static char distro[256] = {'\0'};
+   static bool loaded_data = false;
+
+   (void)this;
+
+   if(!loaded_data)
+   {
+      uname_result = uname(&uname_info);
+      FILE* fp = popen("lsb_release --id --release --codename --short", "r");
+      if(fp != NULL)
+      {
+         if(fread(distro, sizeof(*distro), sizeof(distro), fp))
+         {
+            size_t num_lines = 0;
+            char** d_split = String_split(distro, '\n', &num_lines);
+            distro[0] = '\0';
+            if(num_lines == 3) // lsb_release returns exactly 3 lines
+               snprintf(distro, sizeof(distro), "%s %s (%s)", d_split[0], d_split[1], d_split[2]);
```

Comment:
Not the most robust way to do it, even though it gets the job done.

More conservative approaches would be e.g. (a) using the non-short output and parsing the information based on the prefix (susceptible to changes to those strings) or (b) calling the version with just the short flag and one option each, which makes sure we could detect missing information.

Another issue may arise with the `String_split` and the final `\n`in the process output causing actually 4 strings to be returned by splitting. Thus the check should probably be for 3 lines or more with just caring for the content of the first 3 lines of output.

Reply:
> More conservative approaches would be e.g. (a) using the non-short output and parsing the information based on the prefix (susceptible to changes to those strings) or (b) calling the version with just the short flag and one option each, which makes sure we could detect missing information.

For (a), the [reference page](https://refspecs.linuxfoundation.org/LSB_3.0.0/LSB-Core-generic/LSB-Core-generic.html#LSBRELEASE) gives the format for non-short output (`prefix:\t info`). It doesn't say anything about the prefix strings changing, though.

Calling `lsb_release` thrice as in (b) would avoid any ambiguities regarding the order of the output.

> Another issue may arise with the String_split and the final `\n` in the process output causing actually 4 strings to be returned by splitting. Thus the check should probably be for 3 lines or more with just caring for the content of the first 3 lines of output.

Fixed.